src/module-elasticsuite-catalog/etc/di.xml

```diff
@@ -330,4 +330,41 @@
     <type name="\Magento\Catalog\Api\Data\ProductAttributeInterface">
         <plugin name="resetIndexConfigAfterAttributeSave" type="\Smile\ElasticsuiteCatalog\Plugin\Catalog\Eav\AttributePlugin" />
     </type>
+
+    <type name="Smile\ElasticsuiteCore\Model\Search\Request\Field\Mapper">
+        <arguments>
+            <argument name="fieldNameMapping" xsi:type="array">
+                <item name="price" xsi:type="string">price.price</item>
+                <item name="position" xsi:type="string">category.position</item>
+                <item name="category_ids" xsi:type="string">category.category_id</item>
+            </argument>
+        </arguments>
+    </type>
+
+    <type name="Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product\AttributeList">
+        <arguments>
+            <argument name="requestFieldMapper" xsi:type="object">Smile\ElasticsuiteCatalog\Model\Layer\Filter\Field\Mapper</argument>
+        </arguments>
+    </type>
+    <virtualType name="Smile\ElasticsuiteCatalog\Model\Layer\Filter\Field\Mapper" type="Smile\ElasticsuiteCore\Model\Search\Request\Field\Mapper">
+        <arguments>
+            <argument name="fieldNameMapping" xsi:type="array">
+                <item name="price" xsi:type="string">price.price</item>
+                <item name="categories" xsi:type="string">category.category_id</item>
+            </argument>
+        </arguments>
+    </virtualType>
+
+    <type name="Smile\ElasticsuiteCatalog\Plugin\Search\RequestMapperPlugin">
+        <arguments>
+            <argument name="requestFieldMapper" xsi:type="object">Smile\ElasticsuiteCatalog\Search\Request\Field\Mapper</argument>
+        </arguments>
+    </type>
+    <virtualType name="Smile\ElasticsuiteCatalog\Search\Request\Field\Mapper" type="Smile\ElasticsuiteCore\Model\Search\Request\Field\Mapper">
+        <arguments>
+            <argument name="fieldNameMapping" xsi:type="array">
+                <item name="category_id" xsi:type="string">category.category_id</item>
```

Comment:
Same as above, if "category_id" can exist in the mapping, this should probably exist globally in the object.

Reply:
I will need a bit more time to do some more testing on this one. It may be the case that this one can be removed but I need to do some testing with virtual rules to see what happens.

Reply:
And ... it seems it was added here: https://github.com/Smile-SA/elasticsuite/commit/e5885bb068447e89a6c6acd971931d5d94d9d686#diff-8d2a359b86ebe589e64eb2d7579c6eea4c5679253d3b1ef130eb7e41bec8fbf2

Can you comment on this one?

Reply:
From what I understand the "category_id" mapping was added since GraphQl, and considering the latest changes of the module I think we can add this one `module-elasticsuite-catalog-graph-ql/etc/graphql/di.xml`

`<type name="Smile\ElasticsuiteCatalog\Model\Search\Request\Field\Mapper">
        <arguments>
            <argument name="fieldNameMapping" xsi:type="array">
                <item name="category_id" xsi:type="string">category.category_id</item>
            </argument>
        </arguments>
    </type>`

Reply:
I have done some tests and adding
```xml
<type name="Smile\ElasticsuiteCatalog\Model\Search\Request\Field\Mapper">
    <arguments>
        <argument name="fieldNameMapping" xsi:type="array">
            <item name="category_id" xsi:type="string">category.category_id</item>
        </argument>
    </arguments>
</type>
``` 
to 
`module-elasticsuite-catalog-graph-ql/etc/graphql/di.xml`
does the job, however even if we add it here it will still affect the `$attributeNameMapping` used in
`\Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product\AttributeList::getAttributeCollection`

It does no harm as long as the `category_id` field is located in the array after the 'category_ids' field but if the order is changed it will cause issue because `category_id` does not exist as eav_attribute.

The `$attributeNameMapping` will look like this 
```php
[
    'category_ids' => 'category.category_id',
    'position' => 'category.position',
    'price' => 'price.price',
    **'category_id' => 'category.category_id',**
]
```
and `$fieldNames` like this
```php
[
    ...
    'category.category_id' => 'category_ids'
    'category.position' => 'position'
    'category.is_parent => 'category.is_parent'
    'category.name' => 'category.name'
    ...
]
```

If you have any suggestions how to handle this one let's discuss it, however in my opinion there should be no duplicate mappings for the same attribute.